src/main/scala/mesosphere/marathon/storage/migration/Migration.scala

```diff
@@ -161,20 +157,30 @@ class Migration(
             s" than ${StorageVersions.current.str}."
           throw new MigrationFailedException(msg)
         case Some(version) if version < currentBuildVersion =>
-          if (!backupCreated && config.backupLocation.isDefined) {
-            logger.info("Backup current state")
-            await(backup.backup(config.backupLocation.get))
-            logger.info("Backup finished. Apply migration.")
+          // mark migration as started
+          await(persistenceStore.startMigration())
+          await(runMigrations(version, backupCreated).asTry) match {
+            case Success(result) =>
+              // mark migration as completed
+              await(persistenceStore.endMigration())
+              result
+            case Failure(ex: MigrationCancelledException) =>
+              // mark migration as completed
+              await(persistenceStore.endMigration())
+              throw ex
+            case Failure(ex) =>
+              throw ex
           }
-          val result = await(applyMigrationSteps(version))
-          await(storeCurrentVersion())
-          result
         case Some(version) if version == currentBuildVersion =>
           logger.info("No migration necessary, already at the current version")
           Nil
         case _ =>
           logger.info("No migration necessary, no version stored")
+          // mark migration as started
+          await(persistenceStore.startMigration())
```

Comment:
Since storing storage version is part of the migration?

Reply:
Since the day one, I guess. Since the day this was placed into `Migration.scala`. The flag serves as a guard for any write operation related to versioning and data layouts.